fetch.bs

```diff
@@ -6714,6 +6752,424 @@ agent's <a>CORS-preflight cache</a> for which there is a <a>cache entry match</a
+ <var>onActivatedWithoutTermination</var>.
+
+ <li><p><a for=list>Append</a> <var>deferredRecord</var> to <var>request</var>'s
+ <a for=request>client</a>'s <a>fetch group</a>'s <a for="fetch group">deferred fetch records</a>.
+
+ <li>
+  <p>If <var>activateAfter</var> is non-null, then run the following steps <a>in parallel</a>:</p>
+
+  <ol>
+   <li>
+    <p>The user agent should wait until any of the following conditions is met:
+
+    <ul class=brief>
+     <li><p>At least <var>activateAfter</var> milliseconds have passed.
+
+     <li><p>The user agent has a reason to believe that it is about to lose the opportunity to
+     execute scripts, e.g., when the browser is moved to the background, or when
+     <var>request</var>'s <a for=request>client</a> is a {{Document}} that had a
+     "<code>hidden</code>" <a for=Document>visibility state</a> for a long period of time.
+    </ul>
+
+   <li><p><a data-lt="process a deferred fetch">Process</a> <var>deferredRecord</var>.
+  </ol>
+ </li>
+
+ <li><p>Return <var>deferredRecord</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To compute the <dfn>total request length</dfn> of a <a for=/>request</a> <var>request</var>:
+
+<ol>
+ <li><p>Let <var>totalRequestLength</var> be the <a for=string>length</a> of <var>request</var>'s
+ <a for=request>URL</a>, <a lt="URL serializer">serialized</a> with
+ [=URL serializer/exclude fragment=] set to true.
+
+ <li><p>Increment <var>totalRequestLength</var> by the  <a for=string>length</a> of
+ <var>request</var>'s <a for=request>referrer</a>, <a lt="URL serializer">serialized</a>.
+
+ <li><p><a for=list>For each</a> (<var>name</var>, <var>value</var>) of <var>request</var>'s
+ <a for=request>header list</a>, increment <var>totalRequestLength</var> by <var>name</var>'s
+ <a for="byte sequence">length</a> + <var>value</var>'s <a for="byte sequence">length</a>.
+
+ <li><p>Increment <var>totalRequestLength</var> by <var>request</var>'s <a for=request>body</a>'s
+ <a for=body>length</a>.
+
+ <li><p>Return <var>totalRequestLength</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn>process deferred fetches</dfn> given a <a>fetch group</a> <var>fetchGroup</var>:
+
+<ol>
+ <li><p><a for=list>For each</a> <a for="fetch group">deferred fetch record</a>
+ <var>deferredRecord</var> of <var>fetchGroup</var>'s
+ <a for="fetch group">deferred fetch records</a>, <a>process a deferred fetch</a>
+ <var>deferredRecord</var>.
+</ol>
+</div>
+
+<div algorithm>
+<p>To <dfn>process a deferred fetch</dfn> <var>deferredRecord</var>:
+<ol>
+ <li><p>If <var>deferredRecord</var>'s <a for="deferred fetch record">invoke state</a> is not
+ "<code>pending</code>", then return.
+
+ <li><p>Set <var>deferredRecord</var>'s <a for="deferred fetch record">invoke state</a> to
+ "<code>sent</code>".
+
+ <li><p><a for=/>Fetch</a> <var>deferredRecord</var>'s <a for="deferred fetch record">request</a>.
+
+ <li><p><a>Queue a global task</a> on the <a>deferred fetch task source</a> with
```

Comment:
> Queue a global task on the deferred fetch task source with deferredRecord’s request’s client’s global object to run deferredRecord’s notify invoked.

One question not related to bug: why `notify invoked` needs to be put into global task here? Can't it be executed immediately?

Reply:
No because it's observable from JS

Reply:
Sorry but we don't want it to be observable?

Reply:
It affects the `sent` value of `DeferredFetchResult`

Reply:
Yes. I am confused: we don't want to update `DeferredFetchResult.activated` immediately (as it's at the end of `process ..` steps)?

Reply:
Closing after off-line discussion